pkg/snapshotterutil/sociutil.go

```diff
@@ -0,0 +1,108 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package snapshotterutil
+
+import (
+	"bufio"
+	"os"
+	"os/exec"
+
+	"github.com/sirupsen/logrus"
+)
+
+// CreateSoci creates a SOCI index(`rawRef`)
+func CreateSoci(rawRef string) error {
+	sociExecutable, err := exec.LookPath("soci")
+	if err != nil {
+		logrus.WithError(err).Error("soci executable not found in path $PATH")
+		logrus.Info("you might consider installing soci from: https://github.com/awslabs/soci-snapshotter/blob/main/docs/install.md")
+		return err
+	}
+
+	sociCmd := exec.Command(sociExecutable, []string{"create"}...)
```

Comment:
You should also pass `--namespace`, `--address` (containerd address), and maybe `--timeout`.